### ISSUE TITLE
FraudScenarioTest fails to build metamodel

### ISSUE THREAD

**Issue by bsunderhus** [pextralabs/fraud-scenario#3]
The main FraudScenarioTest is failling right at the moment to evaluate the `KieSession``.

```14:28:12,471  INFO Reflections:229 - Reflections took 1 ms to scan 1 urls, producing 2 keys and 3 values 
14:28:12,472  INFO FraudScenarioTest:58 - Now running data
14:28:12,476 DEBUG SCENELogger:24 - INSERT - co.pextra.fraud.Client@1d05366
14:28:12,476 DEBUG SCENELogger:24 - INSERT - co.pextra.fraud.Device@f143ed
14:28:12,476 DEBUG SCENELogger:24 - INSERT - co.pextra.fraud.AuthToken@1e3314
14:28:12,477  INFO FraudScenarioTest:67 - Final checks
Exception in thread "Thread-0" Exception executing consequence for rule "Session" in co.pextra.fraud: br.ufes.inf.lprm.scene.exceptions.SituationTypeNotFound: co.pextra.fraud.Session not found
    at org.drools.core.runtime.rule.impl.DefaultConsequenceExceptionHandler.handleException(DefaultConsequenceExceptionHandler.java:39)
    at org.drools.core.common.DefaultAgenda.fireActivation(DefaultAgenda.java:1063)
    at org.drools.core.phreak.RuleExecutor.fire(RuleExecutor.java:121)
    at org.drools.core.phreak.RuleExecutor.evaluateNetworkAndFire(RuleExecutor.java:74)
    at org.drools.core.common.DefaultAgenda.fireNextItem(DefaultAgenda.java:970)
    at org.drools.core.common.DefaultAgenda.fireLoop(DefaultAgenda.java:1312)
    at org.drools.core.common.DefaultAgenda.fireUntilHalt(DefaultAgenda.java:1232)
    at org.drools.core.impl.StatefulKnowledgeSessionImpl.fireUntilHalt(StatefulKnowledgeSessionImpl.java:1398)
    at org.drools.core.impl.StatefulKnowledgeSessionImpl.fireUntilHalt(StatefulKnowledgeSessionImpl.java:1377)
    at co.pextra.fraud.RuleEngineThread.run(RuleEngineThread.java:14)
Caused by: br.ufes.inf.lprm.scene.exceptions.SituationTypeNotFound: co.pextra.fraud.Session not found
    at br.ufes.inf.lprm.scene.util.SituationHelper.situationDetected(SituationHelper.java:34)
    at co.pextra.fraud.Rule_Session1742232626.defaultConsequence(Rule_Session1742232626.java:7)
    at co.pextra.fraud.Rule_Session1742232626DefaultConsequenceInvokerGenerated.evaluate(Unknown Source)
    at co.pextra.fraud.Rule_Session1742232626DefaultConsequenceInvoker.evaluate(Unknown Source)
    at org.drools.core.common.DefaultAgenda.fireActivation(DefaultAgenda.java:1052)
    ... 8 more```

**Comment by pereirazc** [comment]
Couldn't reproduce your error, but I got this instead:

```
16:36:25,817 DEBUG SCENELogger:17 - SITUATION ACTIVATION - TYPE: co.pextra.fraud.Session	ID: -1108053688	CAST: {client=co.pextra.fraud.Client@323e8306, location=co.pextra.fraud.Location@4ff735de, token=co.pextra.fraud.AuthToken@57a4d5ee}
java.lang.InstantiationException: co.pextra.fraud.Location
	at java.lang.Class.newInstance(Class.java:427)
	at br.ufes.inf.lprm.scene.model.Snapshot.set(Snapshot.java:97)
	at br.ufes.inf.lprm.scene.model.SituationType.newInstance(SituationType.java:90)
	at br.ufes.inf.lprm.scene.util.SituationHelper.activateSituation(SituationHelper.java:50)
	at br.ufes.inf.lprm.scene.base.Rule_SituationActivation2095507131.defaultConsequence(Rule_SituationActivation2095507131.java:7)
	at br.ufes.inf.lprm.scene.base.Rule_SituationActivation2095507131DefaultConsequenceInvokerGenerated.evaluate(Unknown Source)
	at br.ufes.inf.lprm.scene.base.Rule_SituationActivation2095507131DefaultConsequenceInvoker.evaluate(Unknown Source)
	at org.drools.core.common.DefaultAgenda.fireActivation(DefaultAgenda.java:1052)
	at org.drools.core.phreak.RuleExecutor.fire(RuleExecutor.java:121)
	at org.drools.core.phreak.RuleExecutor.evaluateNetworkAndFire(RuleExecutor.java:74)
	at org.drools.core.common.DefaultAgenda.fireNextItem(DefaultAgenda.java:970)
	at org.drools.core.common.DefaultAgenda.fireLoop(DefaultAgenda.java:1312)
```

**Comment by pereirazc** [comment]
The `snapshop` feature for non-primitive properties needs their classes to have a dummy constructor (no parameters). You should declare a constructor like that for your `Location` class.

It still gives a further error, but it is a `scene-core` bug and it's being solved here: [#12](https://github.com/pextralabs/scene-platform/issues/12)
